dashboard/app/models/concerns/curriculum/course_types.rb

```diff
@@ -53,6 +53,8 @@ def can_be_participant?(user)
     # If unit is in a unit group then decide based on unit group audience
     return unit_group.can_be_participant?(user) if is_a?(Script) && unit_group
 
+    # Signed out users can only use student facing courses
+    return false if !user && participant_audience != 'student'
```

Comment:
what is the experience when a signed-out user goes to a course for facilitators? it would be nice not to show a blank page. I am wondering if we want to redirect to login instead? if we want that, then perhaps we should update the logic for login_required to also require login when `participant_audience != 'student'` in addition / instead?

Reply:
I haven't added the code for locking people out of courses based on the audiences yet however that is coming soon https://github.com/code-dot-org/code-dot-org/pull/43778 . I'm happy to update login_required. Can I do it as a follow up since we aren't yet locking people out of courses?

Reply:
Also which login_required logic were you specifically thinking of?

Reply:
Yes, sounds good. This logic:
* https://github.com/code-dot-org/code-dot-org/blob/1a4319f55da82b336d11d3c1f2fee0b50fd4cf1b/dashboard/app/controllers/script_levels_controller.rb#L119
* https://github.com/code-dot-org/code-dot-org/blob/0c2048a45d6f19b349828872bb7dfd55966e053e/dashboard/app/models/ability.rb#L277-L287